File: app/src/main/java/com/ninhttd/devtest/data/local/SharedPrefsHelper.java
package com.ninhttd.devtest.data.local;

import android.content.Context;
import android.content.SharedPreferences;

import com.ninhttd.devtest.TekoApplication;

public class SharedPrefsHelper {

    public static String PREF_KEY = "teko_key";
    public static String KEY_FIRST_RUN = "first_run";

    private static SharedPreferences sharedPreferences = TekoApplication.getAppContext().getSharedPreferences(SharedPrefsHelper.PREF_KEY, Context.MODE_PRIVATE);
    private static SharedPrefsHelper instance = new SharedPrefsHelper();

    public static SharedPrefsHelper getInstance() {
        return instance;
    }

    public void put(String key, String value) {
        sharedPreferences.edit().putString(key, value).apply();
    }

    public void put(String key, boolean value) {
        sharedPreferences.edit().putBoolean(key, value).apply();
    }

    public String get(String key, String defaultValue) {
        return sharedPreferences.getString(key, defaultValue);
    }

    public Boolean get(String key, boolean defaultValue) {
        return sharedPreferences.getBoolean(key, defaultValue);
    }

    public String get(String key) {
        return get(key, "");
    }

    public void removeAllKey(){
        sharedPreferences.edit().clear().apply();
        getInstance().put(KEY_FIRST_RUN, false);
    }

    public void clearAllKey(){
        removeAllKey();
    }
}

File: app/src/main/java/com/ninhttd/devtest/data/local/SharedPrefsHelper.kt
package com.ninhttd.devtest.data.local

import android.content.Context
import android.content.SharedPreferences

import com.ninhttd.devtest.TekoApplication

class SharedPrefsHelper {

    fun put(key: String, value: String) {
        sharedPreferences.edit().putString(key, value).apply()
    }

    fun put(key: String, value: Boolean) {
        sharedPreferences.edit().putBoolean(key, value).apply()
    }

    @JvmOverloads
    operator fun get(key: String, defaultValue: String = ""): String? {
        return sharedPreferences.getString(key, defaultValue)
    }

    operator fun get(key: String, defaultValue: Boolean): Boolean? {
        return sharedPreferences.getBoolean(key, defaultValue)
    }

    fun removeAllKey() {
        sharedPreferences.edit().clear().apply()
        instance.put(KEY_FIRST_RUN, false)
    }

    fun clearAllKey() {
        removeAllKey()
    }

    companion object {

        var PREF_KEY = "teko_key"
        var KEY_FIRST_RUN = "first_run"

        private val sharedPreferences = TekoApplication.appContext!!.getSharedPreferences(SharedPrefsHelper.PREF_KEY, Context.MODE_PRIVATE)
        val instance = SharedPrefsHelper()
    }
}
